File: backend/Models/DishSet.cs
using System;
using System.Collections.Generic;

namespace food_ordering_server.Models
{
    public class DishSet
    {
        public DishDto Salad { get; set; }
        public DishDto Soup { get; set; }
        public DishDto Roast { get; set; }
        public DishDto Garnish { get; set; }
        public static DishSet CreateFromList(List<DishDto> dishes)
        {
            return new DishSet()
            {
                Salad = dishes.Find((DishDto dish) => dish.Type == 0),
                Soup = dishes.Find((DishDto dish) => dish.Type == 1),
                Roast = dishes.Find((DishDto dish) => dish.Type == 3),
                Garnish = dishes.Find((DishDto dish) => dish.Type == 2),
            };
        }
    }
}

File: frontend/lib/models/dish_set.dart
import 'dish.dart';

class DishSet {
  DishSet({this.salad, this.soup, this.roast, this.garnish});

// ignore_for_file: argument_type_not_assignable, avoid_as
  factory DishSet.fromJson(Map<String, dynamic> json) => json == null
      ? null
      : DishSet(
          salad: Dish.fromJson(json['salad']),
          soup: Dish.fromJson(json['soup']),
          roast: Dish.fromJson(json['roast']),
          garnish: Dish.fromJson(json['garnish']),
        );

  final Dish salad;
  final Dish soup;
  final Dish roast;
  final Dish garnish;

  Iterable<Dish> get dishes => <Dish>[
        if (salad != null) salad,
        if (soup != null) soup,
        if (roast != null) roast,
        if (garnish != null) garnish,
      ];

  static Iterable<Dish> getDishes(DishSet dishSet) => dishSet.dishes;
}
